packages/ui/components/proposal/ProposalYourVote.tsx

```diff
@@ -1,5 +1,27 @@
-export interface ProposalYourVoteProps {}
+import clsx from 'clsx'
 
-export const ProposalYourVote = ({}: ProposalYourVoteProps) => {
-  return <p>placeholder</p>
+export type YourVoteProps = {
+  variant: 'abstain' | 'pending' | 'no' | 'yes'
+}
+
+export const ProposalYourVote = ({ variant }: YourVoteProps) => {
+  return (
+    <div
+      className={clsx(
+        'flex relative justify-center items-center py-1 px-4 w-20 text-sm font-medium rounded-full',
+        variant === 'abstain'
+          ? 'text-secondary bg-primary'
+          : variant === 'pending'
+          ? 'border-2 border-inactive'
+          : variant === 'no'
+          ? 'text-interactive-error bg-[#C73E5914]'
+          : 'text-valid bg-[#39A69914]'
+      )}
+    >
+      {variant[0].toUpperCase() + variant.slice(1)}
+      {variant === 'pending' && (
+        <div className="absolute top-0 right-0 w-2 h-2 bg-[#F53E86] rounded-full"></div>
```

Comment:
Can we use the new color classes (e.g. `text-secondary` should now be `text-text-secondary` IIRC)? See @ezekiiel's amazing wiki page for more info: https://github.com/DA0-DA0/dao-dao-ui/wiki/Implementing-designs-in-the-DAO-DAO-UI

All these hardcoded colors are also probably now covered by the new color names :)

Reply:
Hey looks like some of the colors are not part of the theme so they do have to be hard-coded. If you click on the component and there is a theme color I use it, else I just hardcode it based on Francois's design.